File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
      unordered_map<int,int> mp;
      for(int i=0;i<nums.size();i+=1) {
        int n=nums[i];
        if(mp.find(target-n)!=mp.end()) {
            return {i,mp[target-n]};
        }
        mp[n]=i;
      }
      return {};
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        for(int i=0;i<n;i+=1) {
            for(int j=i+1;j<n;j+=1){
                if(nums[i]+nums[j]==target) return new int[]{i,j};
            }
        }
        return new int[]{};
    }
}